### ISSUE TITLE
[V3] Using fromProviderState on body doesn't work properly

### ISSUE THREAD

**Issue by dpakach** [pact-foundation/pact-js#633]
### Software versions

_Please provide at least OS and version of pact-js_

- **OS**: ubuntu 20.04
- **Consumer Pact library**: "10.0.0-beta.33"
- **Provider Pact library**:
- **Node Version**: v12.21.0

### Issue Checklist

Please confirm the following:

- [x] I have upgraded to the latest
- [ ] I have the read the FAQs in the Readme
- [x] I have triple checked, that there are **no unhandled promises** in my code
- [ ] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [x] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour
fromProviderState on Body should return proper body
eg.
``` js
 provider
      .given("set id", {id: "42"})
      .uponReceiving("a request to get the plain data")
      .withRequest({
        method: "GET",
        path: MatchersV3.fromProviderState("/data/${id}", "/data/42"),
      })
      .willRespondWith({
        status: 200,
        headers: { "Content-Type": 'text/plain; charset=utf-8' },
        body: MatchersV3.fromProviderState("data: testData, id: ${id}", "data: testData, id: 42")
   })

    return provider.executeTest(async (mockserver) => {
      transactionService.setAccountServiceUrl(mockserver.url)
      return transactionService
        .getText(42)
        .then((result) => {
          console.log(result.data)
          expect(result.data).to.equal("data: testData, id: 42")
        })
    })
  })
```
### Actual behaviour

The body returned is not string, rather a pact object
```
 Expected value   "data: testData, id: 42"
    Received:
      {"expression": "data: testData, id: ${id}", "pact:generator:type": "ProviderState", "pact:matcher:type": "type", "value": "data: testData, id: 42"}
    

```

### Steps to reproduce

See the bug reproduced in https://github.com/dpakach/pact-js/pull/2
Test on https://github.com/dpakach/pact-js/pull/2/files#diff-656f935c93a4d4ea910cb4961f3463c434eb6ea7cbe8d51e2c7b0b3fd4c5ae0cR101

**Comment by individual-it** [comment]
`fromProviderState` returns an object  https://github.com/pact-foundation/pact-js/blob/feat/v3.0.0/src/v3/matchers.ts#L451-L455
but maybe we need something like https://github.com/pact-foundation/pact-js/blob/45bb7c46196d71aafbd883ad8d40f7fe3cebf494/src/v3/matchers.ts#L404-L417

**Comment by individual-it** [comment]
it's not limited to `fromProviderState`, I cannot get any matcher working in the body response

**Comment by individual-it** [comment]
adding `body = body.value` after https://github.com/pact-foundation/pact-js/blob/feat/v3.0.0/src/v3/pact.ts#L198 does solve it

**Comment by uglyog** [comment]
Actually, I think this is due to the content type not being JSON, so it not knowing to process the intermediate format

**Comment by individual-it** [comment]
@uglyog you are right setting the headers in the response to `{ "Content-Type": 'application/json; charset=utf-8' }`
make the test pass.
But in this case the content between provider & consumer is not JSON, but as I understand you we need to declare the content to be JSON for pact-js to use it correctly

**Comment by uglyog** [comment]
Sorry, I wasn't saying to do that to fix it, but rather, that is the cause of the issue.

I also can't understand why we are converting the bodies to JSON (https://github.com/pact-foundation/pact-js/blob/feat/v3.0.0/src/v3/pact.ts#L171), when we can just pass them through. I'm going to try change that, so we don't need to use the content type to know if we need to process the body.

**Comment by individual-it** [comment]
should be fixed by #672

**Comment by mefellows** [comment]
v10.0.0-beta.36 is on the way out now.
